File: 122_best_time_to_buy_and_sell_stock_II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size()<2) return 0;
        int max=0;
        int prev=prices[0],pos=prices[0];
        for (int i=1;i<prices.size();i++) {
            if (prices[i]>=prices[i-1]) pos=prices[i];
            else {
                max=max+pos-prev;
                pos=prev=prices[i];
                }
        }
        max=max+pos-prev;
        return max;
    }
};

File: algorithms/python3/122_best_time_to_buy_and_sell_stock_II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprofit=0
        for i in range(1,len(prices)):
            if prices[i]>prices[i-1]:
                maxprofit+=prices[i]-prices[i-1]
        return maxprofit

a=Solution()

print(a.maxProfit([1,2,3,4,5]))